massa-network-worker/src/binders.rs

```diff
@@ -56,21 +58,21 @@ impl WriteBinder {
 
 /// Used to receive and deserialize data.
 pub struct ReadBinder {
-    read_half: ReadHalf,
+    read_half: Resource<ReadHalf, StandardClock>,
     message_index: u64,
     buf: Vec<u8>,
     cursor: usize,
     msg_size: Option<u32>,
 }
 
 impl ReadBinder {
-    /// Creates a new `ReadBinder`.
+    /// Creates a new `ReadBinder` with a bandwidth `limit` in bytes per second.
     ///
     /// # Argument
     /// * `read_half`: reader half.
-    pub fn new(read_half: ReadHalf) -> Self {
+    pub fn new(read_half: ReadHalf, limit: u32) -> Self {
```

Comment:
let's use u64

Reply:
The usage of u64 is tricky because we can't safely change it into a f64 (which is used by the crate). It implies a lot of modifications to integrate Results and Error codes.

An alternative is to directly use a f64 in the config? Is it a good compromise?